pkg/install/billing.go

```diff
@@ -21,10 +29,65 @@ func (i *Installer) createBillingRecord(ctx context.Context) error {
 			Location: i.doc.OpenShiftCluster.Location,
 		},
 	})
-	// If create return a conflict, this means row is already present in database
-	if err, ok := err.(*cosmosdb.Error); ok && err.StatusCode == http.StatusConflict {
-		return nil
+	if err != nil {
+		if err, ok := err.(*cosmosdb.Error); ok && err.StatusCode == http.StatusConflict {
+			i.log.Printf("billing record already present id DB")
+			return nil
+		}
+		return err
+	}
+	// Validate if E2E Feature is registred
+	resource, err := azure.ParseResourceID(billingDoc.Key)
+	if err != nil {
+		return err
 	}
 
-	return err
+	subDocument, err := i.sub.Get(ctx, resource.SubscriptionID)
+	if err != nil {
+		return err
+	}
+	if isSubscriptionRegisteredToE2E(*subDocument) {
+		// We are not failing the operation if we cannot write to e2e storage account, just warning
+		// The blob can only be created when the billing record is also written for the first time
+		errBlob := createE2Eblob(ctx, i.env, *billingDoc, resource)
+		if errBlob != nil {
+			i.log.Warnf("createE2Eblob failed : %s", errBlob.Error())
+		}
+	}
```

Comment:
I am assuming is the create caae. What about update/delete cases?

Reply:
Do we need to update it ?
For deletion i set up a retention policy in the blob storage it self to add a TTL to each blob created

![image](https://user-images.githubusercontent.com/8661347/78155201-273d1c00-740b-11ea-89b5-b4cf80ee5a7c.png)

Reply:
Is this one per Region or one globally? If this is one per region, we should add creation of the StorageAccount to `pkg/deploy/generators/resources.go`

Reply:
we need a deletion time stamp in our current design, no?

Reply:
@julienstroheker We will need to update this blob in the delete operation with the deletion time. Then the e2e can verify the state of the blob, once after create and once after delete.

Reply:
@jim-minter so we need to create the blob when we do create the billing record in cosmos and we also need to update the blob when we do `MarkForDeletion`.

So is it not worth it moving the blob creation/update logique inside `pkg/database/billing.go` to avoid duplicate in term of storage client creation etc... ?

Reply:
I don't think it should go in pkg/database, but I'd be happy to see a wrapper package under pkg/util/billing or something